### PR TITLE
Update dependencies and make required changes

### PR DESCRIPTION
**Pull request by intricate**
Required so that `cardano-node` can build with a newer revision of `cardano-ledger-specs` which contains changes required for https://github.com/input-output-hk/cardano-node/pull/799.

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs [hunk 1]
```diff
@@ -90,10 +94,11 @@ instance Crypto c => HasTxs (ShelleyBlock c) where
 instance Crypto c => ToCBOR (GenTx (ShelleyBlock c)) where
   -- No need to encode the 'TxId', it's just a hash of the 'SL.TxBody' inside
   -- 'SL.Tx', so it can be recomputed.
-  toCBOR (ShelleyTx _txid tx) = toCBOR tx
+  toCBOR (ShelleyTx _txid tx) = wrapCBORinCBOR toCBOR tx
 
 instance Crypto c => FromCBOR (GenTx (ShelleyBlock c)) where
-  fromCBOR = mkShelleyTx <$> fromCBOR
+  fromCBOR = fmap mkShelleyTx $ unwrapCBORinCBOR
+    $ (. Full) . runAnnotator <$> fromCBOR
```
Comment (redxaxder) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs:102]: This was the extent of the serialization changes @mrBliss

Comment (mrBliss) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs:102]: Great :+1:. Can you make a separate PR for this that we can merge after #1947 and #1928?

**Comment by mrBliss** [comment]
> This duplicates #1951. Please wait until the right PRs that bring in these changes piecemeal are merged: #1947, #1928, and a separate one for the Tx annotation changes.

#1947 and #1928 have been merged, so this PR can be rebased and merged. Because the Tx annotation changes are so minor, it's fine for me if they remain part of this PR.

**Comment by intricate** [comment]
> > This duplicates #1951. Please wait until the right PRs that bring in these changes piecemeal are merged: #1947, #1928, and a separate one for the Tx annotation changes.
> 
> #1947 and #1928 have been merged, so this PR can be rebased and merged. Because the Tx annotation changes are so minor, it's fine for me if they remain part of this PR.

Will do.

**Comment by dcoutts** [comment]
bors merge

**Comment by dcoutts** [comment]
There was actually a build failure, but hydra was not reporting to github.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs [hunk 2]
```diff
@@ -39,6 +41,8 @@ import           Ouroboros.Consensus.Shelley.Ledger.Block
 import           Ouroboros.Consensus.Shelley.Ledger.Ledger
 import           Ouroboros.Consensus.Shelley.Protocol
 
+import           Ouroboros.Network.Block (unwrapCBORinCBOR, wrapCBORinCBOR)
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs:44]: Nitpick: can this import be moved between the `Cardano.*` and `Ouroboros.Consensus.*` imports (separated by blank lines)?

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**